ocpmodels/trainers/base_trainer.py

```diff
@@ -246,15 +247,25 @@ def load_model(self):
         if self.logger is not None:
             self.logger.watch(self.model)
 
-    def load_pretrained(self, checkpoint_path=None):
+    def load_pretrained(self, checkpoint_path=None, ddp_to_dp=False):
         if checkpoint_path is None or os.path.isfile(checkpoint_path) is False:
             return False
 
         print("### Loading checkpoint from: {}".format(checkpoint_path))
         checkpoint = torch.load(checkpoint_path)
 
         # Load model, optimizer, normalizer state dict.
-        self.model.load_state_dict(checkpoint["state_dict"])
+        # if trained with ddp and want to load in non-ddp, modify keys from
+        # module.module.. -> module..
+        if ddp_to_dp:
+            new_dict = OrderedDict()
+            for k, v in checkpoint["state_dict"].items():
+                name = k[7:]
```

Comment:
What's the `7` for?

Reply:
Ah, nevermind, got it

Reply:
Leaving this here for reference as well: https://discuss.pytorch.org/t/solved-keyerror-unexpected-key-module-encoder-embedding-weight-in-state-dict/1686/4